### ISSUE TITLE
spider fails when used together with --show_hidden

### ISSUE THREAD

**Issue by damianam** [TACC/Lmod#122]
See example below. `--show_hidden` causes spider to fail to find a package, independently of its visibility. I tried with Lmod 6.0.6 and 6.3. It doesn't seem to be cache related (I tried also with `--ignore_cache`). Is there any other info I can provide to diagnose the problem?

```
user@node> module avail Autoconf

Use "module spider" to find all possible modules.
Use "module keyword key1 key2 ..." to search for all possible modules matching any of the
"keys".

user@node> module --show_hidden avail Autoconf

--------- /usr/local/software/jureca/Stages/2016a/modules/all/Compiler/GCCcore/5.3.0 ---------
   Autoconf/.2.69

Use "module spider" to find all possible modules.
Use "module keyword key1 key2 ..." to search for all possible modules matching any of the
"keys".

user@node> module spider Autoconf
Lmod has detected the following error:  Unable to find: "Autoconf"

user@node> module --show_hidden spider Autoconf
Lmod has detected the following error:  Unable to find: "Autoconf"

```

It also fails to find visible modules when passing `--show_hidden`

```
user@node> module spider CMake

------------------------------------------------------------------------------------------
  CMake:
------------------------------------------------------------------------------------------
    Description:
      CMake, the cross-platform, open-source build system. CMake is a family of tools
      designed to build, test and package software. - Homepage: http://www.cmake.org

     Versions:
        CMake/3.4.3

------------------------------------------------------------------------------------------
  To find detailed information about CMake please enter the full name.
  For example:

     $ module spider CMake/3.4.3
------------------------------------------------------------------------------------------

user@node> module --show_hidden spider CMake
Lmod has detected the following error:  Unable to find: "CMake"
```

**Comment by rtmclay** [comment]
It looks like a bug.  To speed up finding this bug, Would you please put together a test case module tree with only two modules in it:  Say Autoconf/.2.69 and CMake/3.4.3.  Then generate a spider cache for this small module tree.  Finally run 
```module --show_hidden -D spider Autoconf 2> spider.ac.log```
and 
```module --show_hidden -D spider CMake 2> spider.cm.log```

then please send me a tar ball with:

1. The module tree
2.  The spider cache you generated for this small tree
3. The log files: spider.*.log

Thanks,

**Comment by rtmclay** [comment]
With the latest version of Lmod (Try HEAD) I am unable to reproduce your problem:

```
rios(3276)% ml --show_hidden av

-------------------- /home/mclay/t/damain_lmod/modules_test --------------------
   Autoconf/.2.69    CMake/3.4.3

Use "module spider" to find all possible modules.
Use "module keyword key1 key2 ..." to search for all possible modules matching
any of the "keys".

rios(3277)% ml  av             

-------------------- /home/mclay/t/damain_lmod/modules_test --------------------
   CMake/3.4.3

Use "module spider" to find all possible modules.
Use "module keyword key1 key2 ..." to search for all possible modules matching
any of the "keys".
```

**Comment by damianam** [comment]
Thank you Rob. I can confirm that with the latest version it works. It seems the bug was corrected in the transition from 6.3 to 6.3.1:

```
10:55:32 alvarez@jrl01 Lmod-6.3]$ export LMOD_CMD=$HOME/sw/lmod/lmod/6.3/libexec/lmod
10:56:44 alvarez@jrl01 Lmod-6.3]$ ml --show_hidden spider cmake
Lmod has detected the following error:  Unable to find: "cmake"


10:56:45 alvarez@jrl01 Lmod-6.3]$ export LMOD_CMD=$HOME/sw/lmod/lmod/6.3.1/libexec/lmod
10:56:53 alvarez@jrl01 Lmod-6.3]$ ml --show_hidden spider cmake
------------------------------------------------------------------------------------
  CMake: CMake/3.4.3
------------------------------------------------------------------------------------
    Description:
      CMake, the cross-platform, open-source build system. CMake is a family of
      tools designed to build, test and package software. - Homepage:
      http://www.cmake.org

    This module can be loaded directly: module load CMake/3.4.3

    Help:
      CMake, the cross-platform, open-source build system.
       CMake is a family of tools designed to build, test and package software. - Homepage: http://www.cmake.org



(27) 10:56:54 alvarez@jrl01 Lmod-6.3]$
```

Can version 6.3.7 be considered production ready?

**Comment by rtmclay** [comment]
Yes it is production ready.  The only differences between 6.3 and 6.3.7 is a series of bug fixes.
